src/gwt/src/org/rstudio/studio/client/workbench/views/files/ui/FileCommandToolbar.java

```diff
@@ -38,6 +38,39 @@ public FileCommandToolbar(Commands commands, UserPrefs prefs)
 
       addLeftWidget(commands.newFolder().createToolbarButton());
       addLeftSeparator();
+      
+      // New File Menu
+      ToolbarPopupMenu newFileMenu = new ToolbarPopupMenu();
+      newFileMenu.addItem(commands.touchSourceDoc().createMenuItem(false));
+      newFileMenu.addSeparator();
+      newFileMenu.addItem(commands.touchRNotebook().createMenuItem(false));
+      newFileMenu.addItem(commands.touchRMarkdownDoc().createMenuItem(false));
+      newFileMenu.addItem(commands.touchQuartoDoc().createMenuItem(false));
+      newFileMenu.addSeparator();
+      // these two commands will automatically set up files and folders,
+      // so they do not need touch commands
+      newFileMenu.addItem(commands.newRShinyApp().createMenuItem(false));
+      newFileMenu.addItem(commands.newRPlumberDoc().createMenuItem(false));
+      newFileMenu.addSeparator();
+      newFileMenu.addItem(commands.touchTextDoc().createMenuItem(false));
+      newFileMenu.addItem(commands.touchCppDoc().createMenuItem(false));
+      newFileMenu.addItem(commands.touchPythonDoc().createMenuItem(false));
+      newFileMenu.addItem(commands.touchSqlDoc().createMenuItem(false));
+      newFileMenu.addItem(commands.touchStanDoc().createMenuItem(false));
+      newFileMenu.addItem(commands.touchD3Doc().createMenuItem(false));
+      newFileMenu.addSeparator();
+      newFileMenu.addItem(commands.touchSweaveDoc().createMenuItem(false));
+      newFileMenu.addItem(commands.touchRHTMLDoc().createMenuItem(false));
+
+      ToolbarMenuButton newFileButton = new ToolbarMenuButton(
+            "Create New File",
```

Comment:
Nit: To match the "New Folder" button should we call this "New File"?

Reply:
I wanted to distinguish between the existing "New File" operation in the File menu, and this action. I'll go the consistency path -- also fewer words is less UI clutter.

Reply:
This has been changed to "New Blank File" to make it totally clear that it will create a blank file without applying any code templates.